### DISCUSSION TITLE
Expose SceneObjects

### DISCUSSION THREAD

**Discussion by celojevic** [FirstGearGames/FishNet#409] [Feature Request]
When getting that obscure "SceneID # was not found in scene objects" error, it's impossible to tell which objects belongs to the ID unless you make a custom editor that will search the SceneObjects for said ID and return the object. However, SceneObjects is protected, so my suggestion is to make it public or have a public getter that returns it so it can be searched. Thanks

**Comment by FirstGearGames** [comment]
You would only be able to search on the server. If the scene was loaded then the client would not get the error, therefor the id would not show up on clients.

I can expose the collection as readonly public though; should be fine for debugging. Try it out and let me know if it works for you...
rename SceneObjects to `protected Dictionary<ulong, NetworkObject> SceneObjects_Internal = new Dictionary<ulong, NetworkObject>();`
and add this below it `public IReadOnlyDictionary<ulong, NetworkObject> SceneObjects => SceneObjects_Internal;`

I could also consider an alternative of sending scene and object names with the spawn packets so if the sceneId could not be found it will print out the information. This would of course incur bandwidth and small amounts of cpu.

Another option is to serialize the scene name + objects to Ids into a scriptableobject instead of sending but that would be a lot more work and there would be potential for missing information if the SO was deleted and scene was never reopened to serialize it again; so probably not an option.

**Comment by FirstGearGames** [comment]
Added in 3.10.0 
Exposed SceneObjects as well added DebugManager.WriteSceneObjectDetails.
-- awaiting testing @celojevic
